processor/src/test/java/org/mapstruct/ap/test/annotatewith/DeprecateMapper.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.annotatewith;
+
+import org.mapstruct.AnnotateWith;
+import org.mapstruct.Mapper;
+
+/**
+ * @author Ben Zegveld
+ */
+@Mapper
+@AnnotateWith( Deprecated.class )
```

Comment:
The idea was class with annotation without values. Can actually get rid of this one, because later I added `DeprecateAndCustomMapper` as a version with two different annotations, to test that it gets them both. If multiple work then one would also work, so no need for a separate test anymore.